django/http/request.py

```diff
@@ -280,28 +283,35 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.content_type == 'multipart/form-data':
-            if hasattr(self, '_body'):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
-            try:
+        try:
```

Comment:
Is there some history on why this ended up moving inside a `try..catch`? I remember it being mentioned, but it's not clear to me why this behavior would have changed here, and I don't recall the reasoning given at the time.